### PR TITLE
[FLINK-21090][tests] Add IT case for stop-with-savepoint and final checkpoint (1.14)

### PR DESCRIPTION
**Pull request by gaoyunhaii**
This PR picks https://github.com/apache/flink/pull/16773 to release-1.14

### CODE REVIEW

**Comment by gaoyunhaii** [comment]
@flinkbot run azure

File: flink-tests/src/test/java/org/apache/flink/runtime/operators/lifecycle/graph/TestJobBuilders.java [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+                    String unitedSourceLeft = OP_ID_HASH_PREFIX + "1";
+                    String unitedSourceRight = OP_ID_HASH_PREFIX + "2";
+                    String connectedSource = OP_ID_HASH_PREFIX + "3";
+                    String multiSource = OP_ID_HASH_PREFIX + "4";
+                    String mapForward = OP_ID_HASH_PREFIX + "5";
+                    String mapKeyed = OP_ID_HASH_PREFIX + "6";
+                    String mapTwoInput = OP_ID_HASH_PREFIX + "7";
+                    String multipleInput = OP_ID_HASH_PREFIX + "8";
+
+                    // todo: FLIP-27 sources
```
Comment (dawidwys) [inline flink-tests/src/test/java/org/apache/flink/runtime/operators/lifecycle/graph/TestJobBuilders.java:139]: What is the plan for this?

Comment (pnowojski) [reply flink-tests/src/test/java/org/apache/flink/runtime/operators/lifecycle/graph/TestJobBuilders.java:139]: If there is any left over TODO, let's do it after merging @rkhachatryan's PR and let's plan it between me, you @dawidwys  and @gaoyunhaii .

**Comment by dawidwys** [comment]
Merged already.

**Pull request closed without merging**